File: patterns/java/src/refactoring_guru/patterns/chain_of_responsibility/user_authenticator/middleware/UserExistsMiddleware.java
package refactoring_guru.patterns.chain_of_responsibility.user_authenticator.middleware;

import refactoring_guru.patterns.chain_of_responsibility.user_authenticator.server.Server;

/**
 * RU: Конкретный элемент цепи обрабатывает запрос по-своему.
 */
public class UserExistsMiddleware extends Middleware {
    public boolean check(String email, String password) {
        if (!Server.hasEmail(email)) {
            return false;
        }
        return checkNext(email, password);
    }
}

File: patterns/cs/src/RefactoringGuru/ChainOfResponsibility/Example/Middleware/UserExistsMiddleware.cs
namespace RefactoringGuru.ChainOfResponsibility.Example.Middleware
{
    class UserExistsMiddleware : Middleware
    {

        /**
         * RU: Конкретный элемент цепи обрабатывает запрос по-своему.
         */
        public override bool Check(string email, string password)
        {
            if (!Server.HasEmail(email))
            {
                return false;
            }
            return CheckNext(email, password);
        }
    }
}
